### PR TITLE
Bioc updates

### PR DESCRIPTION
**Pull request by glennpj**
Depends on #37090 and #37296

Update Bioconductor packages for Bioconductor 3.17.

### CODE REVIEW

**Comment by tldahlgren** [comment]
Of the 192 packages/versions, #37090 is required for (fetching) the following new versions:

- r-biocio@1.10.0
- r-biocversion@3.17.1
- r-ggtree@3.8.0
- r-gviz@1.44.0

File: var/spack/repos/builtin/packages/r-ggtree/package.py [hunk 1]
```diff
@@ -28,11 +29,13 @@ class RGgtree(RPackage):
     depends_on("r-dplyr", type=("build", "run"))
     depends_on("r-ggplot2@3.0.0:", type=("build", "run"))
     depends_on("r-ggplot2@3.4.0:", type=("build", "run"), when="@3.6.2:")
+    depends_on("r-ggplot2@UNKNOWN:", type=("build", "run"), when="@3.8.0:")
     depends_on("r-magrittr", type=("build", "run"))
     depends_on("r-purrr", type=("build", "run"))
     depends_on("r-rlang", type=("build", "run"))
     depends_on("r-ggfun", type=("build", "run"))
     depends_on("r-ggfun@0.0.6:", type=("build", "run"), when="@3.4.0:")
+    depends_on("r-ggfun@0.0.9:", type=("build", "run"), when="@3.8.0:")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/r-ggtree/package.py:38]: @glennpj I think you're missing the update to `r-ggfun` to `0.0.9`.

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Unless I'm missing something, I think there's one more package that needs to be updated (i.e., `r-ggfun`) in addition to the dependency on the recent `r` package update.

File: var/spack/repos/builtin/packages/r-ggtree/package.py (hunk 1 above)
Comment (glennpj) [reply var/spack/repos/builtin/packages/r-ggtree/package.py:38]: That is part of #37296.

**Comment by tldahlgren** [comment]
After updating the branch (via "Update branch" button) to include the required PRs I can confirm all *BUT* `r-aneufinder@1.28.0`.  When I tried to fetch it, I get:

```
$ spack fetch r-aneufinder@1.28.0 
==> Error: concretization failed for the following reasons:

   1. Cannot satisfy 'r@3.0:'
```

**Comment by glennpj** [comment]
That is odd. I will look into it.

**Comment by glennpj** [comment]
The issue was with the `r-ecp` package. The dependency in the DESCRIPTION file is 
> Depends:	R (≥ 3.00), Rcpp

That got propagated to the Spack recipe but `3.00` is not a valid version. The only thing that depends on `r-ecp` is `r-aneufinder`. I committed a fixed `r-ecp` to solve this.

**Review by tldahlgren — APPROVED** [review]
Confirmed I can now fetch the previously "problematic" package version.

**Pull request merged**